watertap/unit_models/ion_exchange_0D.py

```diff
@@ -0,0 +1,1352 @@
+        # Initialize holdup block
+        flags = blk.properties_in.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args,
+            hold_state=True,
+        )
+        init_log.info("Initialization Step 1 Complete.")
+        # ---------------------------------------------------------------------
+        # Initialize other state blocks
+        # Set state_args from inlet state
+        if state_args is None:
+            state_args = {}
+            state_dict = blk.properties_in[
+                blk.flowsheet().config.time.first()
+            ].define_port_members()
+
+            for k in state_dict.keys():
+                if state_dict[k].is_indexed():
+                    state_args[k] = {}
+                    for m in state_dict[k].keys():
+                        state_args[k][m] = state_dict[k][m].value
+                else:
+                    state_args[k] = state_dict[k].value
+        state_args_out = deepcopy(state_args)
+        for p, j in blk.properties_out.phase_component_set:
+            # comp = blk.properties_out.params.get_component()
+            # if comp.is_solute():
+            if j == blk.config.target_ion:
+                state_args_out["flow_mol_phase_comp"][(p, j)] = (
+                    state_args["flow_mol_phase_comp"][(p, j)] * 1e-3
+                )
+
+        blk.properties_out.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args_out,
+        )
+        blk.state_args_out = state_args_out
+        blk.state_args = state_args
+
+        state_args_waste = deepcopy(state_args)
+
+        for p, j in blk.properties_waste.phase_component_set:
+            if j == "H2O":
+                state_args_waste["flow_mol_phase_comp"][(p, j)] = (
+                    state_args["flow_mol_phase_comp"][(p, j)] * blk.t_waste_param.value
+                )
+
+        blk.state_args_waste = state_args_waste
+
+        blk.properties_waste.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args_waste,
+        )
+
+        init_log.info("Initialization Step 2 Complete.")
+        # ---------------------------------------------------------------------
+        # Solve unit
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+        init_log.info("Initialization Step 3 {}.".format(idaeslog.condition(res)))
+        # ---------------------------------------------------------------------
+        # Release Inlet state
+        blk.properties_in.release_state(flags, outlvl=outlvl)
+        init_log.info("Initialization Complete: {}".format(idaeslog.condition(res)))
+
+    def calculate_scaling_factors(self):
+        super().calculate_scaling_factors()
+
```

Comment:
You may definitely need to revisit the scaling (and initialization) for the model. This seems like a relatively complex model with many variables and I feel like this method for scaling (and initialization) is too simplistic.